samples/propgrid/propgrid.cpp

```diff
@@ -1182,120 +1110,127 @@ void FormMain::PopulateWithStandardItems ()
 
 #ifdef __WXMSW__
-    prop->SetAttribute(wxPG_FILE_SHOW_RELATIVE_PATH,wxT("C:\\Windows"));
-    pg->SetPropertyValue(prop,wxT("C:\\Windows\\System32\\msvcrt71.dll"));
+    prop->SetAttribute(wxPG_FILE_SHOW_RELATIVE_PATH,"C:\\Windows");
+    pg->SetPropertyValue(prop,"C:\\Windows\\System32\\msvcrt71.dll");
 #endif
 
 #if wxUSE_IMAGE
     // An image file property. Arguments are just like for FileProperty, but
     // wildcard is missing (it is autogenerated from supported image formats).
     // If you really need to override it, create property separately, and call
     // its SetWildcard method.
-    pg->Append( new wxImageFileProperty( wxT("ImageFile"), wxPG_LABEL ) );
+    pg->Append( new wxImageFileProperty( "ImageFile", wxPG_LABEL ) );
 #endif
 
-    pid = pg->Append( new wxColourProperty(wxT("ColourProperty"),wxPG_LABEL,*wxRED) );
-    pg->SetPropertyEditor( wxT("ColourProperty"), wxPGEditor_ComboBox );
-    pg->GetProperty(wxT("ColourProperty"))->SetAutoUnspecified(true);
-    pg->SetPropertyHelpString( wxT("ColourProperty"),
-        wxT("wxPropertyGrid::SetPropertyEditor method has been used to change ")
-        wxT("editor of this property to wxPGEditor_ComboBox)"));
+    pid = pg->Append( new wxColourProperty("ColourProperty",wxPG_LABEL,*wxRED) );
+    pg->SetPropertyEditor( "ColourProperty", wxPGEditor_ComboBox );
+    pg->GetProperty("ColourProperty")->SetAutoUnspecified(true);
+    pg->SetPropertyHelpString( "ColourProperty",
+        "wxPropertyGrid::SetPropertyEditor method has been used to change "
+        "editor of this property to wxPGEditor_ComboBox)");
 
-    pid = pg->Append( new wxColourProperty(wxT("ColourPropertyWithAlpha"),
+    pid = pg->Append( new wxColourProperty("ColourPropertyWithAlpha",
                                            wxPG_LABEL,
                                            wxColour(15, 200, 95, 128)) );
-    pg->SetPropertyAttribute(wxT("ColourPropertyWithAlpha"), wxPG_COLOUR_HAS_ALPHA, true);
-    pg->SetPropertyHelpString(wxT("ColourPropertyWithAlpha"),
-        wxT("Attribute \"HasAlpha\" is set to true for this property."));
+    pg->SetPropertyAttribute("ColourPropertyWithAlpha", wxPG_COLOUR_HAS_ALPHA, true);
+    pg->SetPropertyHelpString("ColourPropertyWithAlpha",
+        "Attribute \"HasAlpha\" is set to true for this property.");
 
     //
     // This demonstrates using alternative editor for colour property
     // to trigger colour dialog directly from button.
-    pg->Append( new wxColourProperty(wxT("ColourProperty2"),wxPG_LABEL,*wxGREEN) );
+    pg->Append( new wxColourProperty("ColourProperty2",wxPG_LABEL,*wxGREEN) );
 
     //
     // wxEnumProperty does not store strings or even list of strings
     // ( so that's why they are static in function ).
-    static const wxChar* enum_prop_labels[] = { wxT("One Item"),
-        wxT("Another Item"), wxT("One More"), wxT("This Is Last"), NULL };
+    static wxArrayString enum_prop_labels;
```

Comment:
What was wrong with this array? I'd just keep it as an array of `wxString`s.

Notice that by calling `push_back()` you now add 4 items to it every time this function is called and not just once!

Reply:
I think that declaration of enum_prop_labels as an array of wChar* serves as a demonstration of possibility of initializing property also from this kind of structure.

Reply:
OK, I changed these back to arrays like you asked, but I need to convert them to wxArrayString and wxArrayInt when I pass them to Append and Set.  I deleted the comment above the call to append too which seemed to be referring to the use of a wxChar* (which I now removed).  I stepped through the example and works well.